src/world/GameWorldView.cpp

```diff
@@ -249,6 +249,14 @@ void GameWorldView::DrawGUI(const RoadBuildState& rb, const TerrainRenderer& ter
             MapPoint curPt = terrainRenderer.ConvertCoords(Position(x, y), &curOffset);
             Position curPos = GetWorld().GetNodePos(curPt) - offset + curOffset;
 
+            // we dont own curPt, no need for any rendering...
```

Comment:
I'd combine those 2 checks as it is 1 thing. And maybe add an overload to `IsPlayerTerritory` that takes a player and checks the owner. `gwv.IsPlayerTerritory` can then call that instead so it is similar to `isOwner`. Check all references first, if that does introduce a) unnecessary checks for isOwner and b) callers relying on `gwv.IsPlayerTerritory` checking for ANY player.

Besides that: IMO this is to early. What about a) mouse outside own terriory (mid=22), or selected points outside territory?

Reply:
About IsPlayerTerritory: I have added an overload to World::IsPlayerTerritory, that consumes a playerId.  i think i changed it the way your requested

for the other point, you are right, what was one of the bugs i realized when i edited my last post....